### DISCUSSION TITLE
Extending config from dependency packages

### DISCUSSION THREAD

**Discussion by manojVivek** [tailwindlabs/tailwindcss#13906] [Help]
Hi, I have a monorepo setup with the following file structure:
```
.
└── packages/
    ├── app/
    │   └── web/
    │       └── tailwind.config.js
    └── shared/
        ├── lib-1/
        │   └── tailwind.config.js
        └── lib-2/
            └── tailwind.config.js
```

Here are a few things to note:
1.  `lib-1` and `lib-2` are dependencies of `web`. 
2. Both `lib-1` and `lib-2`'s config files include their own theme extending like custom animations, etc.
3. Apart from `web` consuming the lib projects, they are published to npm and are independently consumed by third-party projects. 

Initially, there was no need for theme extending and there was only one `tailwind.config.js` file under web while was including these lib packages in its `content`. 

Now we have a need for extending the theme, so we introduced their own config files with the theme extensions. But the question is: How do we reliably copy over these theme extensions to `web` and other third-party projects?

We tried something here.

### Attempted Solution
We compiled the `css` file into each of the lib package's `dist` and shipped to be imported in the consuming project.

#### Problem
Alongside the theme overridden styles, these css files also included the styles for default classes. This is both duplication of css content and also conflicting the styles in the consuming projects and breaking some layouts. 

#### Proposed Solution 1
Supporting a way for the lib packages to export their config overrides that can be then included in the consuming project's config somehow? 

#### Proposed Solution 2
Having an option to compile the css files with rules related to the overridden style, so that the duplication side effect in the attempted approach would be solved. 

Please let me know your thoughts on this.

**Comment by wongjn** [comment]
You could consider including the libraries' Tailwind configurations as [presets](https://tailwindcss.com/docs/presets) in `web` and other third-party projects.

**Reply by manojVivek** [reply]
Thanks, I was not aware of `presets`, will check them out.
